src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2907,19 +2917,86 @@ private SocketFactory getSocketFactory() throws IOException {
         return socketFactory;
     }
 
-    // This method contains the old logic of connecting to
-    // a socket of one of the IPs corresponding to a given host name.
-    // In the old code below, the logic around 0 timeout has been removed as
-    // 0 timeout is not allowed. The code has been re-factored so that the logic
-    // is common for hostName or InetAddress.
-    private Socket getDefaultSocket(String hostName, int portNumber, int timeoutInMilliSeconds) throws IOException {
-        // Open the socket, with or without a timeout, throwing an UnknownHostException
-        // if there is a failure to resolve the host name to an InetSocketAddress.
-        //
+    /**
+     * Helper function which traverses through queue of InetAddresses to find a resolved one
+     * 
+     * @param addrq
+     *        Address queue
+     * @param portNumber
+     *        Port Number
+     * @return First resolved queue or unresolved queue if none found
+     * @throws IOException
+     */
+    private InetSocketAddress traverseAddressQueue(Queue<InetAddress> addrq, int portNumber) throws IOException {
```

Comment:
this gets address from Q right? so perhaps getIpAdressFromQueue is more descriptive?